src/result_gen_with_api.py

```diff
@@ -411,37 +443,247 @@ def prepare_scan_weight_result_object(
-            depthmap, height, width = preprocessing.prepare_depthmap(data, width, height, depthScale)
-            depthmap = preprocessing.preprocess(depthmap)
-            depthmaps.append(depthmap)
+        for artifact in artifact_list:
+            mod_artifact = copy.deepcopy(artifact)
+            mod_artifact['download_status'] = False
 
-        depthmaps = np.array(depthmaps)
-        generated_timestamp = datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')
-        weight_predictions = inference.get_weight_predictions_local(depthmaps)
-        print("weight predictions are: ", weight_predictions)
+            mod_artifact['format'] = self.check_artifact_format(artifact['format'])
 
-        weight_result = self.prepare_weight_result_object(weight_predictions.tolist(), generated_timestamp)
-        weight_result_string = json.dumps(weight_result, indent=2, separators=(',', ':'))
-        weight_result_object = json.loads(weight_result_string)
-        if self.api.post_results(weight_result_object) == 201:
-            print("successfully post artifact level weight results: ", weight_result_object)
+            self.add_artifacts_to_format_dictionary(mod_artifact['format'], mod_artifact)
 
-        scan_weight_result = self.prepare_scan_weight_result_object(weight_predictions.tolist(), generated_timestamp)
-        scan_weight_result_string = json.dumps(scan_weight_result, indent=2, separators=(',', ':'))
-        scan_weight_result_object = json.loads(scan_weight_result_string)
-        if self.api.post_results(scan_weight_result_object) == 201:
-            print("successfully post scan level weight results: ", scan_weight_result_object)
+        print("\nPrepared format wise Artifact:\n")
+        pprint.pprint(self.format_wise_artifact)
+
+    def create_scan_dir(self):
+        '''
+        Create a scan dir and format wise dir inside scan dir
+        in which all the artifacts will be downloaded
+        .
+        └── scans
+            ├── 3fa85f64-5717-4562-b3fc-2c963f66afa6
+            │   └── img
+            │       ├── 3fa85f64-5717-4562-b3fc-2c963f6shradul
+            │       ├── 3fa85f64-5717-4562-b3fc-2c963fmayank
+            │       ├── 69869078-33e1-11eb-af63-cf4006664c92
+            │       └── 699b71dc-33e1-11eb-af63-e32a5809de47
+            └── 59560ba2-33e1-11eb-af63-4b01606d9610
+                └── img
+                    ├── 5850e04c-33e1-11eb-af63-4f5622046249
+                    └── 5850e04c-33e1-11eb-af63-4f5622046249_blur.jpg
+        '''
+        if not os.path.isdir(self.scan_dir):
+            os.makedirs(self.scan_dir)
+
+    def create_artifact_dir(self):
+        for artifact_format in self.format_wise_artifact:
+            if not os.path.exists(
+                os.path.join(
+                    self.scan_dir,
+                    artifact_format)):
+                os.makedirs(os.path.join(self.scan_dir, artifact_format))
+
+
+class GenerateResults:
+    pass
+
+
+class MakeResultObjects:
```

Comment:
Is this used anywhere? If not, I suggest to remove this and put it in once it is needed